src/app/users.service.ts

```diff
@@ -1,33 +1,40 @@
 import { Injectable } from '@angular/core';
-import { usersMock } from './users.mock';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/operators';
 import { UserStructure } from './user-structure.model';
+import { Observable } from 'rxjs';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root'                                                                    
 })
 export class UsersService {  
 
-  users: UserStructure[] = usersMock;
+  constructor(private http: HttpClient){}
 
-  getUsers(): UserStructure[]{
-    return this.users;
+  getUsers(): Observable<UserStructure[]>{
+    return this.http.get<UserStructure[]>('http://localhost:8080/users');
```

Comment:
`UserStructure` can be renamed to `User`. Use the same model that you have created earlier

Since you are using the url `http://localhost:8080` many times in this file, you can have it as a global variable something like

```
   const BASE_URL = 'http://localhost:8080';
   const USERS_BASE_URL = `${BASE_URL}/users`;
```

You can use these variables in place of the url strings in the service methods.
The advantage of having this in place is that, later if the `BASE_URL` needs to be changed, then you can just change one variable in your service file and it reflects at all the places instead of you manually making the changes in each file / method.